### PR TITLE
Sentient monkeys are no longer knocked over by mobs in the way.

### LINKED ISSUE #58730: headswapping with a monkeyman gives you the hardstun upon being crossed thing that monkeys have
which is super annoying

thankfully, the monkey AI doesn't move your body around or attack people or anything like that... unless you then become a slimeperson (after the headswap; dunno what happens if you just directly headswap with a monkeyman), create a new body, and then swap into it:
image
the monkey AI in my body woke up after I swapped out of it and then proceeded to space itself out of the shuttle, so I didn't get to do much testing with swapping with it. I think my attempts to swap with it while it was in deep space but not yet dead (while I was on the shuttle) failed, though.

also, incorporeal revenants can cause monkeys to trip by crossing them (or at least their resting bodies). thankfully, the revenant in that round didn't realize that it was able to do that, or it'd have stunlocked me until I died from shocks and its blight.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Title. They used to be immune to this in the past, but then they were refactored into a species and things have changed.

## Why It's Good For The Game
This will fix #58730.

## Changelog
:cl:
fix: Sentient monkeys are no longer subsceptible to being knocked over by mobs in their way.
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
https://www.youtube.com/watch?v=_J6-3l3hCm0

**Comment by LemonInTheDark** [comment]
image

File: code/datums/ai/monkey/monkey_controller.dm [hunk 1]
```diff
@@ -61,6 +62,15 @@ have ways of interacting with a specific mob and control it.
 
 	return ..() //Run parent at end
 
+// Stops sentient monkeys from being knocked over like weak dunces.
+/datum/ai_controller/monkey/on_sentience_gained()
+	..()
+	pawn.RemoveElement(/datum/element/connect_loc_behalf)
```
Comment (LemonInTheDark) [inline code/datums/ai/monkey/monkey_controller.dm:68]: If you're going to remove this manually, please provide more context to the call for the sake of cheapness. also does this even work? the element is on the controller not the monkey

**Comment by Ghommie** [comment]
Thanks for marking this as a draft, I had forgot to do it myself.

**Comment by LemonInTheDark** [comment]
I uh, ok?

File: code/datums/ai/monkey/monkey_controller.dm [hunk 2]
```diff
@@ -47,20 +50,27 @@ have ways of interacting with a specific mob and control it.
...
 
 	return ..() //Run parent at end
 
+// Stops sentient monkeys from being knocked over like weak dunces.
+/datum/ai_controller/monkey/on_sentience_gained()
+	..()
+	RemoveElement(/datum/element/connect_loc_behalf, pawn, loc_connections)
+
+/datum/ai_controller/monkey/on_sentience_lost()
+	..()
```
Comment (Rohesie) [inline code/datums/ai/monkey/monkey_controller.dm:71]:
```suggestion
	. = ..()
	RemoveElement(/datum/element/connect_loc_behalf, pawn, loc_connections)

/datum/ai_controller/monkey/on_sentience_lost()
	. = ..()
```

**Pull request merged**